### PR TITLE
Stochastic matmul verify

### PR DESCRIPTION
**Pull request by andrej**

### CODE REVIEW

**Comment by jgmelber** [comment]
Is this still a WIP or ready to merge?

**Comment by andrej** [comment]
Let me rebase this on refactor-matmul. Then, once refactor-matmul is merged, we can merge this if you are cool with it

**Comment by andrej** [comment]
This also contains the fixes for the verification code on the host (`n_rows` vs `n_cols` confusion). Ready for review! @jgmelber

File: reference_designs/ipu-xrt/matrix_multiplication/test.cpp
Comment (jgmelber) [inline reference_designs/ipu-xrt/matrix_multiplication/test.cpp:201]: Is this reversed?

Comment (andrej) [reply reference_designs/ipu-xrt/matrix_multiplication/test.cpp:201]: Good catch. As it stands, the meaning of this variable is `do_NOT_verify_stochastic`, and I'm printing the wrong log string (says stochastic when it's doing full verification and vice versa). I confused myself. Will flip the condition. Actual verification is correct though

**Review by jgmelber — APPROVED** [review]

**Pull request closed without merging**
